site/app/libraries/NotificationFactory.php

```diff
@@ -234,6 +234,16 @@ public function sendEmails(array $emails, bool $force_secondary = false, bool $s
         $current_user = $this->core->getUser();
         $flattened_emails = [];
         foreach ($emails as $email) {
+            if (!empty($email->getEmailAddress())) {
```

Comment:
Again, `empty()` provides a very loose comparison, and `getEmailAddress()` returns a string, so you can use string comparison instead.

```suggestion
            if ($email->getEmailAddress() !== '') {
```

Reply:
Would this be better than?
`$email->getEmailAddress() !== null && $email->getEmailAddress() !== ''`

Reply:
`getEmailAddress()` returns a string (as per its PHP type annotation), so the null check should not be necessary.